server/sonar-server/src/main/java/org/sonar/server/platform/components/ComponentLevel4.java

```diff
@@ -324,7 +318,7 @@ public ComponentLevel4(ComponentLevel parent, List<Object> level4AddedComponents
   }
 
   @Override
-  public ComponentLevel configure() {
+  protected void configureLevel() {
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)This method has 389 lines, which is greater than the 100 lines authorized. Split it into smaller methods.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS138)